tests/clv/models/test_beta_geo.py

```diff
@@ -40,37 +45,47 @@ def setup_class(cls):
             cls.a_true, cls.b_true, cls.alpha_true, cls.r_true, cls.N, rng=rng
         )
 
-    @pytest.mark.parametrize("a_prior", (None, pm.HalfNormal.dist()))
-    @pytest.mark.parametrize("b_prior", (None, pm.HalfStudentT.dist(nu=4)))
-    @pytest.mark.parametrize("alpha_prior", (None, pm.HalfCauchy.dist(2)))
-    @pytest.mark.parametrize("r_prior", (None, pm.Gamma.dist(1, 1)))
-    def test_model(self, a_prior, b_prior, alpha_prior, r_prior):
-        model = BetaGeoModel(
-            customer_id=self.customer_id,
-            frequency=self.frequency,
-            recency=self.recency,
-            T=self.T,
-            a_prior=a_prior,
-            b_prior=b_prior,
-            alpha_prior=alpha_prior,
-            r_prior=r_prior,
+    @pytest.fixture(scope="class")
+    def data(self):
+        return pd.DataFrame(
+            {
+                "customer_id": self.customer_id,
+                "frequency": self.frequency,
+                "recency": self.recency,
+                "T": self.T,
+            }
         )
 
+    @pytest.fixture(scope="class")
+    def model_config(self):
+        return {
+            "a_prior": {"dist": "halfnormal", "kwargs": {}},
+            "b_prior": {"dist": "halfstudentt", "kwargs": {"nu": 4}},
+            "alpha_prior": {"dist": "halfcauchy", "kwargs": {"beta": 2}},
+            "r_prior": {"dist": "gamma", "kwargs": {"alpha": 1, "beta": 1}},
+        }
+
+    def test_model(self, model_config, data):
+        model = BetaGeoModel(
+            data=data,
+            model_config=model_config,
+        )
+        model.build_model()
         assert isinstance(
             model.model["a"].owner.op,
-            pm.HalfFlat if a_prior is None else type(a_prior.owner.op),
+            type(pm.HalfNormal.dist().owner.op),
         )
         assert isinstance(
             model.model["b"].owner.op,
-            pm.HalfFlat if b_prior is None else type(b_prior.owner.op),
+            type(pm.HalfStudentT.dist(nu=4).owner.op),
         )
         assert isinstance(
             model.model["alpha"].owner.op,
-            pm.HalfFlat if alpha_prior is None else type(alpha_prior.owner.op),
+            type(pm.HalfCauchy.dist(beta=2).owner.op),
         )
         assert isinstance(
             model.model["r"].owner.op,
-            pm.HalfFlat if r_prior is None else type(r_prior.owner.op),
+            type(pm.Gamma.dist(alpha=1, beta=1).owner.op),
```

Comment:
Why isn't this testing the default anymore?

Reply:
the priors aren't passed as a standalone parameters, but as a model_config. the default model config in every class doesn't allow the priors to be None, as it has them always with halfflat dist. So no None case test is needed

Reply:
Sure but we should still test the defaults somewhere even if in a separate test

Reply:
all test_model are now testing for both default and custom config dict scenarios